### PR TITLE
Add GCP installation how to

### PR DESCRIPTION
**Pull request by corvus-ch**

### CODE REVIEW

File: docs/modules/ROOT/pages/explanations/gcp/name_lengths.adoc [hunk 1]
```diff
@@ -0,0 +1,85 @@
+= Names and name length on Google Cloud Platform
+:stem:
+
+[abstract]
+.Abstract
+Node names are limited to a length of 63.
```
Comment (srueg) [inline docs/modules/ROOT/pages/explanations/gcp/name_lengths.adoc:6]: Maybe add a link to here? https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names

File: docs/modules/ROOT/pages/explanations/gcp/name_lengths.adoc [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+If the GCP project name and region are to long, provisioning of nodes will not work and the installer will fail.
```
Comment (srueg) [inline docs/modules/ROOT/pages/explanations/gcp/name_lengths.adoc:7]:
```suggestion
If the GCP project and region name are too long, provisioning of nodes will not work and the installer will fail.
```

File: docs/modules/ROOT/pages/explanations/gcp/name_lengths.adoc [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+When installing a new OpenShift cluster, the installer will create a lot of names automatically.
+Some parts of those names are generated by the installer, others are derived from the underlying cloud.
+
+Let us have a look a how the name of a node is built.
```
Comment (srueg) [inline docs/modules/ROOT/pages/explanations/gcp/name_lengths.adoc:12]:
```suggestion
Let us have a look at how the name of a node is built.
```

File: docs/modules/ROOT/pages/explanations/gcp/name_lengths.adoc [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+`<cluster id>` and `<random>` are generated by the installer.
+`<zone>`, `<region>` and `<project>` are derived from GCP.
+As node names are limited to 63 characters, this can become an issue.
+
+The `<cluster id>` will have a length of twelve characters, `<zone>` is just one characters and `<random>` has a length of five.
+Summing up all the characters that are static and or are generated by the installer, we end up at 37 (see example below).
+This leaves us with 26 characters to be distributed between the project name and the region.
+
+The length of GCP region names vary between eight and 23.
+In the best case, the project can be 18 (stem:[63 - 37 - 8]) character long.
```
Comment (srueg) [inline docs/modules/ROOT/pages/explanations/gcp/name_lengths.adoc:30]:
```suggestion
In the best case, the project can be 18 (stem:[63 - 37 - 8]) characters long.
```

File: docs/modules/ROOT/pages/explanations/gcp/name_lengths.adoc [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+.Example from an actual cluster which exceeded the maximum.
+....
+opensh-5g8lh-w-a-xbtr4.europe-west6-c.c.openshift-4-poc-1.internal
+opensh-5g8lh-w-a-xbtr4.<region>-c.c.<project>.internal
+opensh-5g8lh-w-a-xbtr4.-c.c..internal
+....
+
+[qanda]
+What happens when the node name exceeds 63 characters?::
+The status of the Machine object will be `Provisioned` but not Node object will show up.
```
Comment (srueg) [inline docs/modules/ROOT/pages/explanations/gcp/name_lengths.adoc:44]:
```suggestion
The status of the Machine object will be `Provisioned` but no Node object will show up.
```

File: docs/modules/ROOT/pages/explanations/gcp/name_lengths.adoc [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+The CertificateSigningRequest will not get approved (remains in `Pending`) and a new one will be created every few seconds.
++
+When SSH into the affected VM, one can observe that there is not `/etc/hostname` file and that the hostname is identified as `localhost`.
```
Comment (srueg) [inline docs/modules/ROOT/pages/explanations/gcp/name_lengths.adoc:47]:
```suggestion
When SSH into the affected VM, one can observe that there is no `/etc/hostname` file and that the hostname is identified as `localhost`.
```

File: docs/modules/ROOT/pages/explanations/gcp/name_lengths.adoc [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+ERROR Cluster operator monitoring Degraded is True with UpdatingAlertmanagerFailed: Failed to rollout the stack. Error: running task Updating Alertmanager failed: waiting for Alertmanager Route to become ready failed: waiting for RouteReady of alertmanager-main: no status available for alertmanager-main 
+INFO Cluster operator monitoring Available is False with :  
+INFO Cluster operator monitoring Progressing is True with RollOutInProgress: Rolling out the stack. 
+FATAL failed to initialize the cluster: Some cluster operators are still updating: authentication, console, csi-snapshot-controller, image-registry, ingress, kube-storage-version-migrator, monitoring 
+....
+
+What to do if the length will be exceed and the project name can not be shortened?::
+  Do the cluster setup as normal.
+  The API will come up successfully but the installer will fail.
+  Once this happened, exported the MachineSet objects created by the installer.
```
Comment (srueg) [inline docs/modules/ROOT/pages/explanations/gcp/name_lengths.adoc:84]:
```suggestion
  Once this happened, export the MachineSet objects created by the installer.
```

File: docs/modules/ROOT/pages/how-tos/install/gcp.adoc [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+Region:: <region>
+Base Domain:: <base domain>
+Cluster Name:: <cluster name>
+Pull Secret:: Grab from https://cloud.redhat.com/openshift/install/pull-secret.
+--
++
+.Credentials for Pull Secret
+****
+If the customer brings his own licences, the customer should obtain the Pull Secret using his RedHat account and produce us the Pull Secret.
+For all the other cases, use https://password.vshn.net/cred/detail/2420/.
```
Comment (srueg) [inline docs/modules/ROOT/pages/how-tos/install/gcp.adoc:198]: Should we have references to specific password manager entries in public docs?

Maybe rephrase to
```suggestion
For all the other cases, use the credentials in the password manager.
```

File: docs/modules/ROOT/pages/how-tos/install/gcp.adoc [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+
+5. Review and tweak installer config
++
+Use your editor of choice to review the created installer config.
+Make changes where required.
+
+6. Make a copy of the installer config
++
+Once the installer gets executed, the config will be consumed and deleted.
+If the installer fails and another attempt must be made, having a copy of the config is a good thing to have.
```
Comment (srueg) [inline docs/modules/ROOT/pages/how-tos/install/gcp.adoc:209]:
```suggestion
If the installer fails and another attempt must be made, a copy of the config is a good thing to have.
```

File: docs/modules/ROOT/pages/how-tos/install/gcp.adoc (hunk 8 above)
Comment (chloesoe) [reply docs/modules/ROOT/pages/how-tos/install/gcp.adoc:198]: Discussed that with Christian. Direct links to password.vshn.net would be possible, cause it does not reveal much data, but it is not that "beautiful". 
I would also rephrase it like your suggestion. Perhaps add the title of the password entry, something like:
"For all the other cases, use the credentials in the password manager at 'Red Hat RedHat CCSP Support Account'"

Comment (corvus-ch) [reply docs/modules/ROOT/pages/how-tos/install/gcp.adoc:198]: Using the name as a reference can break badly. The name can change any time. Discussing this with @srueg, we agreed to keep the direct URL. The thinking goes like this:

* How tos within this repository are targeted for the use by VSHN and Puzzle.
* Where procedures differ between those two companies, it is indicated.

We will keep the direct URL but will state that this is for VSHN. Puzzle can then add their place when needed.

File: docs/modules/ROOT/pages/how-tos/install/gcp.adoc [hunk 10]
```diff
@@ -0,0 +1,254 @@
...
+
+Prerequisites:
+* GCP project setup according to above documentation
+* Service Account Key (key.json)
+
+.Required input
+`<base domain>`:: The base domain used to make the cluster available.
+`<cluster name>`::
+The name of the cluster.
+This name will be used as or as part of identifiers.
```
[the remark this answers is not in the document]
Comment (corvus-ch) [reply docs/modules/ROOT/pages/how-tos/install/gcp.adoc:155]: The verbose wording would be

`This name will be used as identifier or as part of identifiers`

Would that make it more understandable?

[the remark this answers is not in the document]
Comment (bliemli) [reply docs/modules/ROOT/pages/how-tos/install/gcp.adoc:155]: Oh I see now, but yes, it would definitely make it more understandable :+1:
